planner/optimize.go

```diff
@@ -162,6 +162,15 @@ func Optimize(ctx context.Context, sctx sessionctx.Context, node ast.Node, is in
 		}
 	}
 
+	// Override the resource group if necessary
+	// TODO: we didn't check the existence of the hinted resource group now to save the cost per query
+	if originStmtHints.HasResourceGroup && variable.EnableResourceControl.Load() {
```

Comment:
I think more readable and can reduce once if judged for normal situations.
```
if originStmtHints.HasResourceGroup {
       if variable.EnableResourceControl.Load() {
       }
       ...
}
```

Reply:
fixed